File: server_py/abi/routes/memory.py
"""Uzun sureli hafiza uc noktalari."""

from __future__ import annotations

from dataclasses import asdict

from fastapi import APIRouter
from fastapi.responses import JSONResponse
from pydantic import BaseModel

from ..providers.memory.file_memory import memory

router = APIRouter()


class FactBody(BaseModel):
    text: str


@router.get("/memory")
async def list_facts() -> dict:
    return {"facts": [asdict(fact) for fact in await memory.list()]}


@router.post("/memory")
async def remember(body: FactBody) -> JSONResponse:
    fact = await memory.remember(body.text)
    if fact is None:
        return JSONResponse({"error": "gecersiz bilgi"}, status_code=400)
    return JSONResponse({"fact": asdict(fact)})


@router.delete("/memory/{fact_id}")
async def forget(fact_id: str) -> dict:
    await memory.forget(fact_id)
    return {"ok": True}


@router.delete("/memory")
async def clear() -> dict:
    await memory.clear()
    return {"ok": True}

File: server/src/routes/memory.ts
import { Router } from 'express';
import { memory } from '../providers/memory/fileMemory.js';

export const memoryRouter: Router = Router();

memoryRouter.get('/memory', async (_req, res) => {
  res.json({ facts: await memory.list() });
});

memoryRouter.post('/memory', async (req, res) => {
  const text = String((req.body as { text?: string }).text ?? '');
  const fact = await memory.remember(text);
  if (!fact) {
    res.status(400).json({ error: 'gecersiz bilgi' });
    return;
  }
  res.json({ fact });
});

memoryRouter.delete('/memory/:id', async (req, res) => {
  await memory.forget(String(req.params.id));
  res.json({ ok: true });
});

memoryRouter.delete('/memory', async (_req, res) => {
  await memory.clear();
  res.json({ ok: true });
});
